scripts/common.py

```diff
@@ -121,6 +121,8 @@
 REDIS_CLI_PATH = os.path.join(
     OPPIA_TOOLS_DIR, 'redis-cli-%s' % REDIS_CLI_VERSION,
     'src', 'redis-cli')
+CLOUD_DATASTORE_EMULATOR_DATA_DIR = os.path.join(
+    OPPIA_TOOLS_DIR, '.config', 'gcloud', 'emulators', 'datastore')
```

Comment:
Does this actually need to be in a top-level directory called .config? If not, can we name it something more specific?

What exactly will this directory contain?

Reply:
This is basically a cache for the Datastore emulator (analogous to the storage used by the `--save_datastore` start.py flag for maintaining state in between runs).

How's `../oppia_tools/cloud_datastore_emulator_cache/`?

Reply:
Could you maybe do two things: (a) change the name to tmp_cloud_datastore_emulator_cache/ so that people know it can be deleted without consequence (if that's true), and (b) add a comment here explaining what that directory contains? Then it should be fine IMO.

@DubeySandeep would you agree?

Reply:
Not sure whether it's a good idea to have datastore (cache) in the oppia_**tools** dir? Maybe have it in separate dir (in ../oppia/) as we have for karma_coverage_report, release deploy dir, etc?

Is the cache dir going to get deleted while flushing the stored data in the datastore? (or it's just the internal data/files inside the dir is going to get deleted?)

Also, I'm not sure but the flush happens when we run the start.py script, right? If yes, then I think the cache dir will exist all the time, right? (If yes, then I'm not sure whether it's nice to have tmp_ prefix)

Reply:
Suggestion for naming:

If the flushing happens after we stop the script then:
   `tmp_cloud_datastore_emulator_cache` (I would have preferred tmp_datastore_cache but considering other dir name very prices like karma_coverage_report instead of coverage_report I think `tmp_cloud_datastore_emulator_cache` is better)

If the flushing happens after we run the start script then:
   `cloud_datastore_emulator_cache` (As this won't be a tmp dir i.e, it will exist in the system all/most of the time)

One more question, is this a cache or the actual data being actively used by the emulator?

Reply:
+1 to having an entirely separate directory (sorry for not thinking of that before!)

Reply:
OK, the flushing will happen before the emulator starts (just like `start.py --save_datastore=false`), and I've moved it to the `oppia/cloud_datastore_emulator_cache`.

Reply:
@DubeySandeep -- you didn't mean `../oppia/` right? karma_coverage_report, release deploy dirs etc. are siblings of the oppia folder, not in the oppia folder.

Reply:
NP, I've just made it a sibling directory.